NBitcoin/PubKey.cs

```diff
@@ -348,8 +348,9 @@ public override bool Equals(object obj)
 			PubKey item = obj as PubKey;
 			if(item == null)
 				return false;
-			return ToHex().Equals(item.ToHex());
+			return Equals(item);
 		}
+		public bool Equals(PubKey pk) => pk != null && ToBytes().SequenceEqual(pk.ToBytes());

```

Comment:
Can you use `NBitcoin.Utils.ArrayEqual(ToBytes(), pk.ToBytes())` it is a bit better.